keras/utils/vis_utils.py

```diff
@@ -135,16 +147,45 @@ def model_to_dot(model,
             node_key = layer.name + '_ib-' + str(i)
             if node_key in model._network_nodes:
                 for inbound_layer in node.inbound_layers:
-                    if not expand_nested or not (
-                            isinstance(inbound_layer, Wrapper) and
+                    inbound_layer_id = str(id(inbound_layer))
+                    # if inbound_layer is not Model or wrapped Model
+                    if (not (expand_nested and (
```

Comment:
You should put this series of checks into a utility function, because 1) you do it multiple times, 2) it would improve code readability.

Reply:
I've organized my codes to make it more readable. As for the unit tests, previous PR [#11431](https://github.com/keras-team/keras/pull/11431) had added some tests. I think that's sufficent. Is it OK?